addons/mail/static/src/new/core/persona_service.js

```diff
@@ -32,21 +33,7 @@ export class PersonaService {
             // Get reactive version.
             persona = this.store.personas[localId];
         }
-        const {
-            id = persona.id,
-            name = persona.name,
-            im_status = persona.im_status,
-            email = persona.email,
-            type,
-        } = data;
-        Object.assign(persona, {
-            localId,
-            id,
-            name,
-            im_status,
-            email,
-            type,
-        });
+        assignDefined(persona, { ...data, localId });
```

Comment:
It looks like localId should only be assigned on the "create" case and not here, removing the need to spread the data.

Reply:
Not a big deal I would say, because this is implementation details of `insert` method.

The tradeoff with current rudimentary implementation of dataclasses would imply adding 1 line of code in the branching of new. It's not necessarily a good tradeoff.

Very low priority aspect, especially when it's solved by improving dataclasses further (e.g. `readonly` modifiers on some fields)